test/assertions_test.rb

```diff
@@ -440,4 +440,17 @@ def test_assert_when_using_no_prefix
       StatsD.increment("incr", no_prefix: true)
     end
   end
+
+  def test_client_propagation_to_expectations
+    foo_1_metric = StatsD::Instrument::Expectation.increment("foo")
+    @test_case.assert_statsd_expectations([foo_1_metric]) do
+      StatsD.increment("foo")
+    end
+
+    client = StatsD::Instrument::Client.new(prefix: "prefix")
+    foo_2_metric = StatsD::Instrument::Expectation.increment("foo", client: client)
+    @test_case.assert_statsd_expectations([foo_2_metric]) do
+      StatsD.increment("prefix.foo")
```

Comment:
This actually brings up an interesting question...

**Should** the expectation even be able to detect this metric, since it was made by the "default" client?

I guess maybe the answer is that the expectation doesn't actually assert which client it came from, it uses the client to detect configuration and asserts about that 🤔